### PR TITLE
chore(ai-autofix): Update autofix payload + use code mapping repos

### PR DESCRIPTION
**Pull request by jennmueng**
Update the ai autofix request payload to send the organization, project, list of repos associated, and remove the hardcoded support for only `getsentry/sentry`

Retrieves relevant repos for a project from the code mappings.

Adds a feature flag in lieu of the hardcode.

### CODE REVIEW

File: src/sentry/api/endpoints/group_ai_autofix.py [hunk 1]
```diff
@@ -49,41 +47,26 @@ class GroupAiAutofixEndpoint(GroupEndpoint):
...
-            for commit in commits:
-                if commit.repository_id == sentry_repo.id:
-                    return commit
-        except Repository.DoesNotExist:
-            logger.exception(
-                "No getsentry/sentry repo found for organization",
-                extra={"group.id": group.id, "group.organization.id": group.organization.id},
-            )
-            pass
+            if len(repo_name_sections) == 2:
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/group_ai_autofix.py:60]: what happens if its not 2? should we handle that case?

File: src/sentry/api/endpoints/group_ai_autofix.py [hunk 2]
```diff
@@ -161,6 +146,11 @@ def post(self, request: Request, group: Group) -> Response:
         if not request.user.is_authenticated:
             raise PermissionDenied(detail="You must be authenticated to perform this action.")
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/group_ai_autofix.py:147]: you shouldn't have to do this? i think you can delete

File: src/sentry/api/endpoints/group_ai_autofix.py (hunk 1 above)
Comment (jennmueng) [reply src/sentry/api/endpoints/group_ai_autofix.py:60]: I think github only does 2 sections? But we can make it so it supports as many sections as it comes in but the first section is the owner and the remaining is the name

File: src/sentry/api/endpoints/group_ai_autofix.py [hunk 3]
```diff
@@ -158,8 +142,10 @@ def post(self, request: Request, group: Group) -> Response:
             ],
         }
 
-        if not request.user.is_authenticated:
-            raise PermissionDenied(detail="You must be authenticated to perform this action.")
+        if not features.has("projects:ai-autofix", group.project):
+            return self._respond_with_error(
+                group, metadata, "AI Autofix is not enabled for this project.", 403
```
Comment (trillville) [inline src/sentry/api/endpoints/group_ai_autofix.py:147]: Are we consistently calling it "AI Autofix"

**Review by trillville — APPROVED** [review]

File: src/sentry/api/endpoints/group_ai_autofix.py (hunk 3 above)
Comment (jennmueng) [reply src/sentry/api/endpoints/group_ai_autofix.py:147]: yes, on the product itself it has been

File: src/sentry/api/endpoints/group_ai_autofix.py [hunk 4]
```diff
@@ -49,41 +46,26 @@ class GroupAiAutofixEndpoint(GroupEndpoint):
...
+        for repo_config in repo_configs:
+            repo: Repository = repo_config.repository
+            repo_name_sections = repo.name.split("/")
 
-        commits: list[Commit] = list(
-            Commit.objects.filter(
-                id__in=ReleaseCommit.objects.filter(release__in=releases_query_set).values("commit")
-            )
-        )
+            if len(repo_name_sections) > 1:
```
Comment (corps) [inline src/sentry/api/endpoints/group_ai_autofix.py:59]: Sanity check -- why does this predicate work?  Maybe a comment?

**Review by corps — APPROVED** [review]

**Pull request merged**